File: core/src/main/java/de/qabel/core/config/factory/DefaultContactFactory.java
package de.qabel.core.config.factory;

import de.qabel.core.config.Contact;
import de.qabel.core.crypto.QblECPublicKey;
import de.qabel.core.drop.DropURL;

import java.util.Collection;

public class DefaultContactFactory implements ContactFactory {
    @Override
    public Contact createContact(QblECPublicKey publicKey, Collection<DropURL> dropUrls, String alias) {
        return new Contact(alias, dropUrls, publicKey);
    }
}

File: core/src/main/java/de/qabel/core/config/factory/DefaultContactFactory.kt
package de.qabel.core.config.factory

import de.qabel.core.config.Contact
import de.qabel.core.crypto.QblECPublicKey
import de.qabel.core.drop.DropURL

class DefaultContactFactory : ContactFactory {
    override fun createContact(publicKey: QblECPublicKey, dropUrls: Collection<DropURL>, alias: String): Contact {
        return Contact(alias, dropUrls, publicKey)
    }
}
